### PR TITLE
[SYCL][Graph] Bug fixes for hanging tests and urCommandBufferAppendKernelLaunchExp 

### PR DESCRIPTION
**Pull request by julianmi**
This patch targets to address two issues found in the testing:
- Test hangs: This patch serializes the graph submissions in the tests to prevent hangs when submitting multiple graphs.
- Issue adding a kernel to the command buffer in `urCommandBufferAppendKernelLaunchExp`.

### CODE REVIEW

File: sycl/test-e2e/Graph/submission_while_executing.cpp [hunk 1]
```diff
@@ -0,0 +1,96 @@
+// REQUIRES: cuda || level_zero, gpu
```
Comment (mfrancepillois) [inline sycl/test-e2e/Graph/submission_while_executing.cpp:1]: cuda support for sycl-graph is not yet available on the xmain branch.

// REQUIRES: cuda || level_zero, gpu 
should be 
// REQUIRES: level_zero, gpu

**Review by mfrancepillois — APPROVED** [review]

File: sycl/test-e2e/Graph/submission_while_executing.cpp (hunk 1 above)
Comment (julianmi) [reply sycl/test-e2e/Graph/submission_while_executing.cpp]: Good catch, thanks. This is fixed.

**Comment by aelovikov-intel** [comment]
I've been cancelling all runs that run without https://github.com/intel/llvm/pull/11240 because we still haven't recovered our Win CI to its desired reliability. I hoped that people would merge origin/sycl instead of just re-running tasks that were explicitly cancelled.

I'm going to let this one go through, as I wanted to start running CI on a case-by-case basis inspecting runners after each task anyway.

**Review by smaslov-intel — APPROVED** [review]
sycl/plugins LGTM

**Pull request merged**